### PR TITLE
fix: fix bfd static route tests

### PR DESCRIPTION
**Pull request by cyw233**
### Description of PR

Summary:
Add fixes and improvements to `bfd/test_bfd_static_route.py`.

To prevent this PR from being too large, it only contains initial fixes/optimizations to `bfd/test_bfd_static_route.py`. I will create another PR later to include things like parametrizing IPv4 and IPv6 for each test, moving fixtures and static methods out of `BfdBase` class and etc.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
To make sure `bfd/test_bfd_static_route.py` can run effectively on all T2 devices
#### How did you verify/test it?
I have run all test cases in this file in a T2 chassis device and can confirm they all passed.

### CODE REVIEW

File: tests/bfd/bfd_base.py [hunk 1]
```diff
@@ -14,7 +16,11 @@ def list_to_dict(self, sample_list):
         data_rows = sample_list[3:]
         for data in data_rows:
             data_dict = {}
-            data = data.encode("utf-8").split()
+            if sys.version_info.major < 3:
+                data = data.encode("utf-8").split()
+            else:
+                data = data.split()
```
Comment (cyw233) [inline tests/bfd/bfd_base.py:22]: Making this compatible for both Python 2 and 3

File: tests/bfd/conftest.py [hunk 2]
```diff
@@ -47,8 +48,8 @@ def bfd_cleanup_db(
     )
     for dut in duts:
         assert wait_until(
-            45, 2, 0, check_orch_cpu_utilization, dut, orch_cpu_threshold
-        ), "Orch CPU utilization {} > orch cpu threshold {} before starting the test".format(
+            120, 4, 0, check_orch_cpu_utilization, dut, orch_cpu_threshold
+        ), "Orch CPU utilization {} > orch cpu threshold {} after the test".format(
```
Comment (cyw233) [inline tests/bfd/conftest.py:52]: I found that it (especially the RP) will sometimes take more time to settle down after rebooting or config reloading, so I bumped the timeout to 120s here.

File: tests/bfd/conftest.py [hunk 3]
```diff
@@ -111,4 +113,32 @@ def bfd_cleanup_db(
             else request.config.dst_asic
         )
         for interface in selected_interfaces:
-            bfd_base_instance.interface_cleanup(dut, asic, interface)
+            ensure_interface_is_up(dut, asic, interface)
```
Comment (cyw233) [inline tests/bfd/conftest.py:116]: The `bfd_base_instance.interface_cleanup()` function was absent originally. Based on the tests and also the logging message above, I assume we want to ensure all the interfaces are up after tests on portchannels or portchannel members, so I added a `ensure_interface_is_up()` function below.

File: tests/bfd/test_bfd_static_route.py [hunk 4]
```diff
@@ -13,8 +16,7 @@
 
 
 class TestBfdStaticRoute(BfdBase):
-    test_case_status = True
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:16]: `test_case_status` seems to be unnecessary.

File: tests/bfd/test_bfd_static_route.py [hunk 5]
```diff
@@ -44,7 +46,7 @@ def modify_bfd_sessions(self, duthosts, bfd_base_instance):
                 ]
                 for asic in asics:
                     assert wait_until(
-                        300,
+                        600,
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:49]: Bump the timeout here to give DUTs more time to shut down all BFD sessions

File: tests/bfd/test_bfd_static_route.py [hunk 6]
```diff
@@ -68,7 +70,7 @@ def modify_bfd_sessions(self, duthosts, bfd_base_instance):
...
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:73]: Bump the timeout here to give DUTs more time to start up all BFD sessions

File: tests/bfd/test_bfd_static_route.py [hunk 7]
```diff
@@ -145,17 +147,12 @@ def test_bfd_with_lc_reboot_ipv4(
...
-                "BGP check failed, not all BGP sessions are up. Failed: {}".format(
-                    failed
-                )
-            )
+        assert wait_until(
+            300,
+            10,
+            0,
+            lambda: check_bgp_status(request),
+        )
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:155]: I found that we need more time to start up BGP sessions, especially in the tests where make changes to RP, so I changed all BGP checks in this test file to a timeout function. Non-RP related tests will have a timeout of 300s, while RP related tests will have a timeout of 600s

File: tests/bfd/test_bfd_static_route.py [hunk 8]
```diff
@@ -107,7 +109,7 @@ def test_bfd_with_lc_reboot_ipv4(
             dst_dut_nexthops,
             src_prefix,
             dst_prefix,
-        ) = self.select_src_dst_dut_with_asic(
+        ) = select_src_dst_dut_with_asic(
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:112]: I moved all static functions in this class to `bfd_helpers.py`

File: tests/bfd/test_bfd_static_route.py [hunk 9]
```diff
@@ -218,8 +210,9 @@ def test_bfd_with_lc_reboot_ipv4(
             ),
         )
 
-    def test_bfd_static_route_deletion_ipv4(
+    def test_bfd_with_lc_reboot_ipv6(
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:213]: I adjusted the order of test cases so that the same test for IPv4 and IPv6 can sit next to each other.

**Comment by wsycqyz** [comment]
Can you write down what test you did in the "How did you verify/test it?" section?

**Comment by cyw233** [comment]
> Can you write down what test you did in the "How did you verify/test it?" section?

Hey @wsycqyz, I have updated the PR description. Cheers

**Review by wsycqyz — APPROVED** [review]

**Review by abdosi — APPROVED** [review]

**Pull request merged**